### PR TITLE
Rewrite :control

### PR DESCRIPTION
**Pull request by evanultra01**
Control has been broken since 2015. Here's a fix.
FUN FACT: The old version of this command caused me to get banned by a game's anticheat system when a game moderator used it on me for a joke.

### CODE REVIEW

File: MainModule/Server/Commands/Moderators.lua [hunk 1]
```diff
@@ -3127,40 +3127,35 @@ return function(Vargs, env)
...
-								p.Transparency = 1
-							elseif p:IsA("Accoutrement") then
-								p:Destroy()
+						plr.Character = newCharacterModel
+
+						newCharacterModel:PivotTo(oldCFrame)
+						newCharacterModel.Parent = workspace
+
+						-- hacky way to fix other people being unable to see animations.
+						for _ = 1, 2 do
```
Comment (P3tray) [inline MainModule/Server/Commands/Moderators.lua:3151]: wtf is going on here

**Review by P3tray — APPROVED** [review]
we need a :uncontrol now

File: MainModule/Server/Commands/Moderators.lua [hunk 2]
```diff
@@ -3127,40 +3127,35 @@ return function(Vargs, env)
...
 							end
 						end
-						v.Character.Parent = plr.Character
-						--v.Character.Humanoid.Changed:Connect(function() v.Character.Humanoid.PlatformStand = true end)
+
+						server.Remote.Send(plr, "Function", "SetView", plr.Character.Humanoid)
+						server.Remote.Send(v, "Function", "SetView", plr.Character.Humanoid)
 					end
 				end
 			end
```
Comment (Sceleratls) [inline MainModule/Server/Commands/Moderators.lua:3161]: Your code is very slow and unoptimized, I have applied the following changes to optimize the complication time of the code. Please approve the following and I will merge this PR.

```suggestion
local Humanoid = v.Character.Humanoid;
local HumanoidDescription = Humanoid:GetAppliedDescription() or service.Players:GetHumanoidDescriptionFromUserId(v.UserId);
local newCharacterModel: Model = service.Players:CreateHumanoidModelFromDescription(HumanoidDescription, v.Character.Humanoid.RigType);
local Animate: BaseScript = newCharacterModel.Animate;
newCharacterModel.Humanoid.DisplayName = Humanoid.DisplayName;
newCharacterModel.Name = v.Name;
local oldCFrame = v.Character and v.Character:GetPivot() or CFrame.new();
if plr.Character then;
plr.Character = nil;
v.Character = nil;
end;
for _, p in plr.Character:GetChildren() do;
if p:IsA("BasePart") then;
p.Transparency = 1;
elseif p:IsA("Accoutrement") then;
p:Destroy();
plr.Character = newCharacterModel;
newCharacterModel:PivotTo(oldCFrame);
newCharacterModel.Parent = workspace;
-- hacky way to fix other people being unable to see animations.;
for _ = 1, 2 do;
if Animate then;
Animate.Disabled = not Animate.Disabled;
end;
end;
v.Character.Parent = plr.Character;
--v.Character.Humanoid.Changed:Connect(function() v.Character.Humanoid.PlatformStand = true end);
server.Remote.Send(plr, "Function", "SetView", plr.Character.Humanoid);
server.Remote.Send(v, "Function", "SetView", plr.Character.Humanoid);
end;
end;
end;
```

**Comment by DimensionalFusion** [comment]
You stink @Sceleratls

**Comment by WaviestBalloon** [comment]
> we need a :uncontrol now

:ref

**Review by ccuser44 — APPROVED** [review]

**Pull request merged**